### PR TITLE
[IN-3268] transaction_type and kind enums

### PR DESCRIPTION
**Pull request by jbutton**
enums for line_items.`transaction_type` and `kind` in responses:
Review renewal:
Screenshot 2024-01-05 at 12 23 59
Subscription preview:
Screenshot 2024-01-05 at 12 27 41
Allocation preview:
Screenshot 2024-01-05 at 12 28 14

### CODE REVIEW

File: components/schemas/Allocation-Preview-Line-Item.yaml [hunk 1]
```diff
@@ -3,8 +3,17 @@ type: object
 properties:
   transaction_type:
     type: string
+    enum:
```
Comment (maciej-nedza) [inline components/schemas/Allocation-Preview-Line-Item.yaml:6]: Please put the enums in separate files and use $refs

File: components/schemas/Allocation-Preview-Line-Item.yaml [hunk 2]
```diff
@@ -3,8 +3,17 @@ type: object
...
+      - charge
+      - credit
   kind:
     type: string
+    enum:
```
Comment (maciej-nedza) [inline components/schemas/Allocation-Preview-Line-Item.yaml:11]: so it looks like its the same enum as in `Component-Kind.yaml`. 
Then we can reuse it here:
```
  kind:
    $ref: "./Component-Kind.yaml"
```

File: components/schemas/Renewal-Preview-Line-Item.yaml [hunk 3]
```diff
@@ -3,8 +3,19 @@ type: object
 properties:
   transaction_type:
     type: string
+    enum:
```
Comment (maciej-nedza) [inline components/schemas/Renewal-Preview-Line-Item.yaml:6]: After it gets extracted to the separate file, use same ref as for `transaction_type` from `components/schemas/Allocation-Preview-Line-Item.yaml`

Comment (jbutton) [reply components/schemas/Renewal-Preview-Line-Item.yaml:6]: from my research allocation preview will have a different set than renewal preview and subscription preview. Will use $ref when have final version ready 👍

**Review by maciej-nedza — APPROVED** [review]
LGTM

File: components/schemas/Line-Item-Transaction-Type.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+title: Line Item Kind
```
Comment (maciej-nedza) [inline components/schemas/Line-Item-Transaction-Type.yaml:1]: Correct titles so there are no duplicates

**Review by macbury — APPROVED** [review]

**Pull request merged**